workflows/metaval.nf

```diff
@@ -64,18 +72,42 @@ workflow METAVAL {
     /*
         SUBWORKFLOW: TAXID_READS
     */
-    TAXID_READS (
-    ch_extract_reads.reads,
-    ch_extract_reads.kraken2_taxpasta,
-    ch_extract_reads.kraken2_result,
-    ch_extract_reads.kraken2_report,
-    ch_extract_reads.centrifuge_taxpasta,
-    ch_extract_reads.centrifuge_result,
-    ch_extract_reads.centrifuge_report,
-    ch_extract_reads.diamond_taxpasta,
-    ch_extract_reads.diamond_tsv,
-    )
-    ch_versions            = ch_versions.mix( TAXID_READS.out.versions )
+    if ( params.perform_extract_reads ) {
+        TAXID_READS (
+        ch_extract_reads.reads,
+        ch_extract_reads.kraken2_taxpasta,
+        ch_extract_reads.kraken2_result,
+        ch_extract_reads.kraken2_report,
+        ch_extract_reads.centrifuge_taxpasta,
+        ch_extract_reads.centrifuge_result,
+        ch_extract_reads.centrifuge_report,
+        ch_extract_reads.diamond_taxpasta,
+        ch_extract_reads.diamond_tsv,
+        )
+        ch_versions            = ch_versions.mix( TAXID_READS.out.versions )
+    }
+
+    /*
+        SUBWORKFLOW: Screen pathogens
+    */
+    ch_reference = Channel.fromPath( params.pathogens_genome, checkIfExists: true)
+        .map{ file -> [ [ id: file.baseName ], file ] }
+    // Short reads
+    if ( params.perform_screen_pathogens ) {
```

Comment:
Maybe the `perform_screen_pathogens` could have a more descriptive name to include that short reads is implied?

Reply:
`perform_screen_pathogens`  turns on the screen pathogens step for both short and long reads. If there are only short reads, then subworkflow `LONGREAD_SCREENPATHOGEN` will be ignore.